scripts/Test-LCOW-UVM.ps1

```diff
@@ -0,0 +1,149 @@
+#ex: .\scripts\Test-LCOW-UVM.ps1 -vb -Action Bench -BootFilesPath C:\ContainerPlat\LinuxBootFiles\ -MountGCSTest -Count 2 -Benchtime '3s'
+# benchstat via  `go install golang.org/x/perf/cmd/benchstat@latest`
+
+[CmdletBinding()]
+param (
+    [ValidateSet('Test', 'Bench', 'List', 'Shell')]
+    [alias('a')]
+    [string]
+    $Action = 'Bench',
+
+    [string]
+    $Note = '',
+
+    # test parameters
+    [int]
+    $Count = 1,
+
+    [string]
+    $BenchTime = '5s',
+
+    [string]
+    $Timeout = '10m',
+
+    [alias('tv')]
+    [switch]
+    $TestVerbose,
+
+    [string]
+    $Run = '',
+
+    [string]
+    $CodePath = '.',
+
+    [string]
+    $OutDirectory = '.\test\results',
+
+    # uvm parameters
+
+    [string]
+    $BootFilesPath = 'C:\ContainerPlat\LinuxBootFiles',
+
+    [ValidateSet('vhd', 'initrd')]
+    [string]
+    $BootFSType = 'vhd',
+
+    [switch]
+    $DisableTimeSync,
+
+    # gcs test/container options
+
+    [string]
+    $ContainerRootFSMount = '/run/rootfs',
+
+    [string]
+    $ContainerRootFSPath = (Join-Path $BootFilesPath 'rootfs.vhd'),
+
+    [string]
+    $GCSTestMount = '/run/bin',
+
+    [string]
+    $GCSTestPath = '.\bin\test\gcstest',
+
+    [switch]
+    $MountGCSTest,
+
+    [string]
+    $Feature = ''
+)
+
+Import-Module ( Join-Path $PSScriptRoot Testing.psm1 ) -Force
+
+$CodePath = Resolve-Path $CodePath
+$OutDirectory = Resolve-Path $OutDirectory
+$BootFilesPath = Resolve-Path $BootFilesPath
+$ContainerRootFSPath = Resolve-Path $ContainerRootFSPath
+$GCSTestPath = Resolve-Path $GCSTestPath
+
+$shell = ( $Action -eq 'Shell' )
+
+if ( $shell ) {
+    $cmd = 'ash'
+} else {
+    $date = Get-Date
+    $waitfiles = "$ContainerRootFSMount"
+    $gcspath = 'gcstest'
+    if ( $MountGCSTest ) {
+        $waitfiles += ",$GCSTestMount"
+        $gcspath = "$GCSTestMount/gcstest"
+    }
+
+    $pre = "wait-paths -p $waitfiles -t 5 ; " + `
```

Comment:
Is this impacted at all by https://github.com/microsoft/hcsshim/pull/1474?

Reply:
This relies on the [wait-paths tool](https://github.com/microsoft/hcsshim/tree/main/cmd/hooks/wait-paths), so it should be fine